app/build.gradle

```diff
@@ -29,12 +32,17 @@ android {
 }
 
 dependencies {
-    compile fileTree(include: ['*.jar'], dir: 'libs')
-    testCompile 'junit:junit:4.12'
-    compile 'com.android.support:appcompat-v7:23.1.1'
-    compile 'com.android.support:design:23.1.1'
-    compile 'com.android.support:recyclerview-v7:23.1.1'
-    compile 'eu.chainfire:libsuperuser:1.0.0.+'
-    compile 'com.google.zxing:core:3.2.1'
-    compile 'org.wordpress:passcodelock:1.+'
+    implementation 'com.android.support:appcompat-v7:23.4.0'
```

Comment:
Updating those dependencies gives me a crash on Android 9 when manually adding a password to a network:

```
3115-3115/be.brunoparmentier.wifikeyshare.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: be.brunoparmentier.wifikeyshare.debug, PID: 3115
    java.lang.ClassCastException: be.brunoparmentier.wifikeyshare.ui.activities.WifiNetworkActivity$NfcFragment cannot be cast to be.brunoparmentier.wifikeyshare.ui.activities.WifiNetworkActivity$QrCodeFragment
        at be.brunoparmentier.wifikeyshare.ui.activities.WifiNetworkActivity$7.onClick(WifiNetworkActivity.java:240)
        at android.view.View.performClick(View.java:6597)
        at android.view.View.performClickInternal(View.java:6574)
        at android.view.View.access$3100(View.java:778)
        at android.view.View$PerformClick.run(View.java:25881)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6649)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:826)
```

The fragments are probably not well managed in `WifiNetworkActivity.java`.

Reply:
Should be fixed now.